include/odp/api/spec/packet_io_types.h

```diff
@@ -1015,6 +1022,32 @@ typedef struct odp_pktin_vector_capability_t {
 
 } odp_pktin_vector_capability_t;
 
+/** Packet output capabilities regarding packets with shared data
+ *
+ *  These capabilities indicate whether packets that reference other packets
+ *  or are referenced by other packets can be sent so that they get consumed by
+ *  the packet output. When such a packet is consumed by the packet output at
+ *  the completion of packet transmit, the packet is handled the same as in
+ *  odp_packet_free() (i.e. internal reference counts are updated and the packet
+ *  segments are put back to the packet pool when there are no more references
+ *  left).
+ *
+ *  If these capabilities are not set, then the respective packet types may be
+ *  sent only if the pktout supports the dont_free option and the option is set
+ *  for the packets being sent. In that case the packets being sent are not
+ *  consumed nor modified.
+ *
+ *  See odp_pktio_capability_t::free_ctrl
+ *  See odp_packet_free_ctrl_set()
+ *  See odp_packet_ref() and odp_packet_ref_static()
+ */
+typedef struct {
+	uint8_t static_ref  :1; /**< Static packet references can be consumed */
+	uint8_t referenced  :1; /**< Referenced packets can be consumed */
+	uint8_t referencing :1; /**< Referencing packets can be consumed */
+
+} odp_pktout_packet_ref_capability_t;
```

Comment:
Better have the same style used elsewhere in file (spec above the field, newlines after fields):
```C
typedef struct {
	/** Static packet references can be consumed */
	uint8_t static_ref  :1;

	/** Referenced packets can be consumed */
	uint8_t referenced  :1;

	/** Referencing packets can be consumed */ 
	uint8_t referencing :1;

} odp_pktout_packet_ref_capability_t;
```

Same comment for the traffic manager capa structure.

Reply:
Argee with Tuomas that we should prefer /** style. I think we have cleaned up /**< comments away over the years, but there are still some left.

Reply:
Changed in v6.